js/main.js

```diff
@@ -1,91 +1,110 @@
 /* global data */
 /* exported data */
 
-// User can create new entry *START*
 var $title = document.querySelector('#note-title');
 var $urlInput = document.querySelector('#url');
 var $notes = document.querySelector('#entry-notes');
 var $submit = document.querySelector('form');
+var $ul = document.querySelector('ul');
+var $editEntryFormTitles = document.querySelectorAll('form>h1');
 
 function setImage(event) {
-
   var image = document.querySelector('#form-image');
   image.setAttribute('src', $urlInput.value);
 }
 
-$urlInput.addEventListener('blur', setImage);
+$urlInput.addEventListener('input', setImage);
 
 function submitForm(event) {
   event.preventDefault();
-  data.entries.push({
-    title: $title.value,
-    photoURL: $urlInput.value,
-    notes: $notes.value,
-    entryId: data.nextEntryId
-  });
-  data.nextEntryId++;
-  $submit.reset();
   var entryList = document.querySelector('ul');
-  entryList.prepend(newEntryCreation(data.entries.length - 1));
-
   var noEntries = document.querySelector('[name="no-entry"');
+  if (data.editing !== null) {
+    data.entries.splice(data.editing, 1, {
+      title: $title.value,
+      photoURL: $urlInput.value,
+      notes: $notes.value,
+      entryId: data.editing
+    });
+
+    entryList.prepend(newEntryCreation(data.editing));
```

Comment:
You shouldn't be adding a new DOM tree for the edited entry to the top of the list, you should be replacing the original DOM tree for this entry with the updated DOM tree after editing